integrations/SD_Lon/calculate_primary.py

```diff
@@ -0,0 +1,321 @@
+import os
+import logging
+import argparse
+import datetime
+
+import sd_common
+import sd_payloads
+
+from os2mo_helpers.mora_helpers import MoraHelper
+
+MORA_BASE = os.environ.get('MORA_BASE', None)
+
+logger = logging.getLogger("updatePrimaryEngagements")
+LOG_LEVEL = logging.DEBUG
+LOG_FILE = 'calculate_primary.log'
+
+
+class MOPrimaryEngagementUpdater(object):
+    def __init__(self):
+        self.helper = MoraHelper(hostname=MORA_BASE, use_cache=False)
+        self.org_uuid = self.helper.read_organisation()
+
+        self.mo_person = None
+
+        # Keys are; fixed_primary, primary no_sallery non-primary
+        self.eng_types = sd_common.engagement_types(self.helper)
+        self.primary = [
+            self.eng_types['fixed_primary'],
+            self.eng_types['primary'],
+            self.eng_types['no_sallery']
+        ]
+
+    def set_current_person(self, cpr=None, uuid=None, mo_person=None):
+        """
+        Set a new person as the current user. Either a cpr-number or
+        an uuid should be given, not both.
+        :param cpr: cpr number of the person.
+        :param uuid: MO uuid of the person.
+        :param mo_person: An already existing user object from mora_helper.
+        :return: True if current user is valid, otherwise False.
+        """
```

Comment:
Bør man ikke explicit enforce reglen - at man kun må angive en af dem:
if uuid and cpr:
    raise ValueError("only one of (uuid, cpr) may be given at any time")